In [1]:
# Load the tidyverse package and read the players data set.
library(tidyverse)
players <- read_csv("data/players.csv")
players

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors
Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


experience subscribe
1   Pro         TRUE    
2   Veteran     TRUE    
3   Veteran    FALSE    
4   Amateur     TRUE    
5   Regular     TRUE    
6   Amateur     TRUE    
7   Regular     TRUE    
8   Amateur    FALSE    
9   Amateur     TRUE    
10  Veteran     TRUE    
11  Veteran     TRUE    
12  Pro         TRUE    
13  Beginner    TRUE    
14  Amateur     TRUE    
15  Veteran     TRUE    
16  Amateur    FALSE    
17  Beginner    TRUE    
18  Amateur     TRUE    
19  Amateur     TRUE    
20  Regular     TRUE    
21  Regular     TRUE    
22  Amateur    FALSE    
23  Beginner    TRUE    
24  Beginner    TRUE    
25  Amateur     TRUE    
26  Regular     TRUE    
27  Regular     TRUE    
28  Veteran    FALSE    
29  Amateur     TRUE    
30  Veteran    FALSE    
⋮   ⋮          ⋮        
167 Amateur    FALSE    
168 Beginner   FALSE    
169 Veteran     TRUE    
170 Veteran    FALSE    
171 Pro        FALSE    
172 Beginner   FALSE    
173 Veteran     TRUE    
174 Regular    FALSE    
175 Veteran     TRUE    
176 Amateur     TRUE    
177 Pro        FALSE    
178 Veteran    FALSE    
179 Amateur     TRUE    
180 Veteran     TRUE    
181 Amateur    FALSE    
182 Amateur     TRUE    
183 Pro        FALSE    
184 Amateur     TRUE    
185 Pro        FALSE    
186 Regular    FALSE    
187 Veteran     TRUE    
188 Amateur     TRUE    
189 Beginner    TRUE    
190 Amateur    FALSE    
191 Amateur     TRUE    
192 Amateur     TRUE    
193 Veteran    FALSE    
194 Amateur    FALSE    
195 Amateur    FALSE    
196 Pro         TRUE    
    hashedEmail                                                     
1   f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d
2   f3c813577c458ba0dfef80996f8f32c93b6e8af1fa939732842f2312358a88e9
3   b674dd7ee0d24096d1c019615ce4d12b20fcbff12d79d3c5a9d2118eb7ccbb28
4   23fe711e0e3b77f1da7aa221ab1192afe21648d47d2b4fa7a5a659ff443a0eb5
5   7dc01f10bf20671ecfccdac23812b1b415acd42c2147cb0af4d48fcce2420f3e
6   f58aad5996a435f16b0284a3b267f973f9af99e7a89bee0430055a44fa92f977
7   8e594b8953193b26f498db95a508b03c6fe1c24bb5251d392c18a0da9a722807
8   1d2371d8a35c8831034b25bda8764539ab7db0f63938696917c447128a2540dd
9   8b71f4d66a38389b7528bb38ba6eb71157733df7d1740371852a797ae97d82d1
10  bbe2d83de678f519c4b3daa7265e683b4fe2d814077f9094afd11d8f217039ec
11  7a4686586d290c67179275c7c3dfb4ea02f4d317d9ee0e2cee98baa27877a875
12  4caa42e1b20511552434978171dcf7283fb6eb857eb871327e39bfc328002309
13  4bed6f49b672974f706aa7f020b4f29020326bf594520593557cd381a51559b8
14  3caa832978e0596779f4ee7c686c4592fb6de893925025d25bf26be9a8cd03f5
15  6f9acf8ea9956fe817895c78d10e1e25c11aba335a451e5b0322b073ee5008c4
16  2313a06afe47eacc28ff55adf6f072e7d12b0d12d7cbae29656f9029ec72bb53
17  9db37a1d381697f6c83e3b97b276e38bb9d9c811ba4596edb75d4dbfea6415b0
18  a175d4741dc84e6baf77901f6e8e0a06f54809a34e6b5211159bced346f7fb3e
19  ab1f44f93c3b828f55458971db393052d9711df3e0e7ff69540bfebfcec555ff
20  20ae3f8b5323674fe6e537bc91928841cfc2a00c1b36522fd25370bb6fa89d36
21  bc704ff2bc676dbf48ee41b9e11481c1387bf758ad318f2428f336e3fecc6660
22  200db0417ee70863c7e2a0f214c737476fb86de5869d0b907a077d726357e687
23  f9ac013b2f0bc2bd4928a6a0fc8a0aae8b5c4f2670cf9e18f547b7e72034c965
24  89c8e07d6a2aa102d9cd05086d51b2e41195c9c807dd976a06f4262ff36d0cca
25  119f01b9877fc5ea0073d05602a353b91c4b48e4cf02f42bb8d661b46a34b760
26  5baba1651a0b92788bc0d6dcdf00be64af1cf9f0015bbeed204ddabd1d872e83
27  50cec73a632e4be7b155c3df5165de28b2c5baf092abcb7f509f5645776a56c2
28  f8acd26a5e452b682b9f8b20108caef1ee2e745efe08e95806980bd2388a3fea
29  4b01bce3f141289709e8278b02ba5d2aaa7105d7ccb9c7deb37670a80e332774
30  951e54f7376e2b2f0915e9e3646c701af4a2fe839385b13f9e0084b08f8070bd
⋮   ⋮                                                               
167 7096b30972a32029244fc6171f21bc5996b51b8d75df7a94f003dead428988f9
168 42eafe96ed5c1684e3b5cc614d1b01a117173d3ec6898abcf1177d9822bd39f7
169 9ba67e127e4e70d909f6237644bb4674b17bc6ae5c9e735f88cfc50e22b499c6
170 4bfad3613c71ace05644bf210195d9fb0d3

In [2]:
# Find the number of observations in the players data set.
count(players)

n  
1 196

In [3]:
# Produce summary statistics by finding the number of players in each level of gaming experience.
players_experience <- players |>
    group_by(experience) |>
    count()
players_experience

experience n 
1 Amateur    63
2 Beginner   35
3 Pro        14
4 Regular    36
5 Veteran    48

In [4]:
# Produce summary statistics by finding the proportion of players subscribed and unsubscribed to the game-related newsletter.
players_subscribe <- players |>
    group_by(subscribe) |>
    count()
players_subscribe

subscribe n  
1 FALSE      52
2  TRUE     144

In [5]:
# Produce summary statistics by determining if any emails are identical.
players_email <- players |>
    group_by(hashedEmail) |>
    count()
players_email

hashedEmail                                                      n
1   0088b5e134c3f0498a18c7ea6b8d77b4b0ff1636fc93355ccc95b45423367832 1
2   055d1a2002f6308a95b03b8d17af28f43e9f91ca8b1e1fc439bf4d08b22e7c0f 1
3   060aca80f8cfbf1c91553a72f4d5ec8034764b05ab59fe8e1cf0eee9a7b67967 1
4   0ce7bfa910d47fc91f21a7b3acd8f33bde6db57912ce0290fa0437ce0b97f387 1
5   0d4d71be33e2bc7266ee4983002bd930f69d304288a8663529c875f40f1750f3 1
6   0d70dd9cac34d646c810b1846fe6a85b9e288a76f5dcab9c1ff1a0e7ca200b3a 1
7   11006065e9412650e99eea4a4aaaf0399bc338006f85e80cc82d18b49f0e2aa4 1
8   119f01b9877fc5ea0073d05602a353b91c4b48e4cf02f42bb8d661b46a34b760 1
9   11bf6125c4264b3a8f3bffa57b33bd598e2ea1ecd6331ad3bc54aa88e22db035 1
10  126ec350278422ba60ad575d230c19522fd568094b7c0fc17f8cc27fa4c810ab 1
11  1683a3e0aed65119f83540274ff6f965fdf66890613a80f88572dfb1d9f1d2ad 1
12  171d3c323c6430ffe0722c931f4cced5c48935f6804e0ff250964e4c41da55b1 1
13  18936844e06b6c7871dce06384e2d142dd86756941641ef39cf40a9967ea14e3 1
14  1a2b92f18f36b0b59b41d648d10a9b8b20a2adff550ddbcb8cec2f47d4d881d0 1
15  1d2371d8a35c8831034b25bda8764539ab7db0f63938696917c447128a2540dd 1
16  1edebbf13898dbe99da5ee743b1fd9cbfdc6d69f1180dd6261da662bacb1a7e5 1
17  200db0417ee70863c7e2a0f214c737476fb86de5869d0b907a077d726357e687 1
18  20ae3f8b5323674fe6e537bc91928841cfc2a00c1b36522fd25370bb6fa89d36 1
19  22a78bfd44e46931261d06864b92cf9a8fd77fb02bec66cb71646dd9da222a0d 1
20  2313a06afe47eacc28ff55adf6f072e7d12b0d12d7cbae29656f9029ec72bb53 1
21  23fe711e0e3b77f1da7aa221ab1192afe21648d47d2b4fa7a5a659ff443a0eb5 1
22  24d4892c2c3ce11d3e54c3bf31ee218901cac7a2f564911d67424c8fdf0874fe 1
23  25879aecc205544bc6505f9faf768356e0a3b7126057308ae3235f8a514e9a62 1
24  26362473ca8fab46358dbab9bcbcf7114bbcbadef90e6bf9212093f74bde9fe3 1
25  2917e2dacafed92d017b9f356ce90a442bea6e4ad59dbbb484a920398acb8974 1
26  2cbbfa45451a38b85663b7a9a3fffed12af1ba063ec03613d4c9e3873090cd05 1
27  2cfed571797b66cc810c32562fc5b0f70b5bec0f5250799245527135bbf9662e 1
28  32f753965c7b684cdda8cb19ff9c8734a9afa41a4ddf4a1dfb9594097d78e1a7 1
29  336adf9b9070abb44eaec22a48b04d8cda6cf0c98a46eef37779a1dcb6f5b4bb 1
30  360c682d9a10b9010e0d357a2aac315ab31c296367c236de5993b647541868ff 1
⋮   ⋮                                                                ⋮
167 e3e1bef1ce259c995332635c0ec630a575713451cbe742d7ca93e4c03d0826df 1
168 e3f0ad9aadd27f3d1d9197e58546d045018daa767675033d9af9a2b74e763739 1
169 e44041459da2102dc20147ed6f0db4753547be66fc4ddea037bafca64004de2d 1
170 e47ae138603b9a7d6e9b97554bac2daf0477b30e0a4f29ed9d3deacc3c7ec155 1
171 e74c60a92c0100e7240be56d66969db85856152b048c639331a3595f901461ae 1
172 e8df74a0605dedc2d6ee2d2651019f9006965f8b70223d5bcf9c26337eb6df05 1
173 ed3e6d043dee74fad932be44d0600f4502055198c511927b0c906f50b57d9582 1
174 eeb961422a3f9841018d4eb729ddc40ef103cc1205577d908b5ee43d4810bee6 1
175 f174555ae3ff613d5d0b98f83f26f689619d92ef272f28f3aa236003d3dbb159 1
176 f19e136ddde68f365afc860c725ccff54307dedd13968e896a9f890c40aea436 1
177 f1b432523542f90c61176a555ccb2144468d76c91a32d74082ab8c101f9d25b6 1
178 f2826fb8dbce4d450348f99cb27ade184b713998d9679780442efaaf218038f2 1
179 f3c813577c458ba0dfef80996f8f32c93b6e8af1fa939732842f2312358a88e9 1
180 f459ce1862ec9a43b6c714810390dfd632ed104eb11378b3a2a134b1b107bc75 1
181 f4f29f2465743ead8c3ce02fc84542afe33ae3c95ef56a42b8f7785550fe9e4a 1
182 f58aad5996a435f16b0284a3b267f973f9af99e7a89bee0430055a44fa92f977 1
183 f6b7fa3cfaeb2f8d2cedf287cac000e219be75a0278039edc4c69eaf96250683 1
184 f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d 1
185 f707d141be44b2bf963eb171dbd14599059b4dda09aa002ba95ead4fa67273a7 1
186 f7875ae87a61632030d5c4029ee8cf081be7047b2b4a9c04ab5a706f59786197 1
187 f8acd26a5e452b682b9f8b20108caef1ee2e745efe08e95806980bd2388a3fea 1
188 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc 1
189 f9ac013b2f0bc2bd4928a6a0fc8a0aae8b5c4f2670cf9e18f547b7e72034c965 1
190 fa7d496b2f74c51ec70395bd8397b49f97a3ce8d7ba7e08887d82ec0008137a7 1
191 fb0005032363ec3b30c2ae66

In [6]:
# Calculate summary statistics by finding the average hours played by the players.
players_avg_hours <- players |>
    summarize(played_hours = mean(played_hours, na.rm = TRUE))
players_avg_hours

players_max_hours <- players |>
    summarize(played_hours = max(played_hours, na.rm = TRUE))
players_max_hours

players_min_hours <- players |>
    summarize(played_hours = min(played_hours, na.rm = TRUE))
players_min_hours

played_hours
1 5.845918

played_hours
1 223.1

played_hours
1 0

In [7]:
# Produce summary statistics by determining the occurence of each name.
players_name <- players |>
    group_by(name) |>
    count()
players_name

name      n
1   Aarav     1
2   Aaron     1
3   Adrian    1
4   Ahmed     1
5   Aiden     1
6   Akio      1
7   Akira     1
8   Alaric    1
9   Alex      1
10  Amelia    1
11  Amelie    1
12  Anastasia 1
13  Andy      1
14  Arash     1
15  Aria      1
16  Ariana    1
17  Arlo      1
18  Asher     1
19  Atlas     1
20  Aurora    1
21  Ayman     1
22  Bailey    1
23  Balthazar 1
24  Bella     1
25  Bianca    1
26  Bijan     1
27  Blake     1
28  Bodhi     1
29  Caden     1
30  Callum    1
⋮   ⋮         ⋮
167 Sidney    1
168 Sophia    1
169 Soren     1
170 Suki      1
171 Taylor    1
172 Thatcher  1
173 Théo      1
174 Tristan   1
175 Umar      1
176 Vasco     1
177 Vaughn    1
178 Violet    1
179 Vivian    1
180 Vivienne  1
181 Will      1
182 Willem    1
183 Winslow   1
184 Winston   1
185 Wren      1
186 Xander    1
187 Xenos     1
188 Xia       1
189 Yosef     1
190 Yvette    1
191 Zane      1
192 Zara      1
193 Zayn      1
194 Zelda     1
195 Ziad      1
196 Zoe       1

In [8]:
# Produce summary statistics by finding the genders present in the players data set.
players_gender <- players |>
    group_by(gender) |>
    count()
players_gender

gender            n  
1 Agender             2
2 Female             37
3 Male              124
4 Non-binary         15
5 Other               1
6 Prefer not to say  11
7 Two-Spirited        6

In [9]:
# Calculate summary statistics by finding the average age of players in the data set.
players_age <- players |>
    summarize(Age = mean(Age, na.rm = TRUE))
players_age

players_max_age <- players |>
    summarize(Age = max(Age, na.rm = TRUE))
players_max_age

players_min_age <- players |>
    summarize(Age = min(Age, na.rm = TRUE))
players_min_age

Age     
1 21.13918

Age
1 58

Age
1 9

# Data Description
The data presented was collected by a research group in computer science at UBC led by Frank Wood, used to predict usage of a Minecraft server by gathering data on the players and sessions involved. This project aims to curate recruitment efforts and ensure ample resources for the attracted players. 


### Players Data Set
The players data set includes a list of all unique players with data about each player. After reading the data set, I am able
to identify that there are 7 variables and 196 observations. There are a variety of data types, including character, logical and double. 
- experience(chr): the level of experience of a player, includes amateur, beginner, regular, veteran, and pro
- subscribe(lgl): the status of subscription to the game-related newsletter, TRUE for subscribed and FALSE for unsubscribed
- hashedEmail(chr): an encrypted version of an email address used to protect the user's personal information, for identification
- played_hours(dbl): the number of hours played on Minecraft
- name(chr): the player's name
- gender(chr): the player's gender
- Age(dbl): the player's age

**Experience Levels**

The table shows that nearly a third of the players are classified as amateur and the least number of players are considered pro level.
| Experience | Count |
| ---------- | ----- |
| Amateur    | 63    |
| Beginner   | 35    |
| Regular    | 36    |
| Veteran    | 48    |
| Pro        | 14    |

**Subscription Status**

About 73% of players are subscribed to the Minecraft newsletter.
| Subscribe | Count |
| --------- | ----- |
| FALSE     | 52    |
| TRUE      | 144   |

**Hashed Email**

Since a hashed email is a one-way encrypted version of an email address, converted to protect the user's personal information, each player has a unique code (no repeats). Thus, this might be used instead to identify players instead of numbering them 1 through 196.

**Hours**

The average hours played by the users in the data set is 5.85 hours, with a maximum of 223.1 hours and a minimum of 0 hours.
| Average Hours | Max Hours | Min Hours |
| ------------- | --------- | --------- |
| 5.85          | 233.1     | 0         |

**Name**

Similar to the hashedEmail variable, the player's names can be used as identification since each is unique to the player with no repeats. 

**Gender**

As provided in the table below, majority of players are male with some females. A smaller number of people are agender, two-spirited, and non-binary.
| Gender            | Count |
| ----------------- | ----- |
| Agender           | 2     |
| Female            | 37    |
| Male              | 124   |
| Non-binary        | 15    |
| Other             | 1     |
| Prefer not to say | 11    |
| Two-spirited      | 6     |

**Age**

The average age of players in the data set is 21.14 years old, with a maximum age of 58 and a minimum age of 9.
| Average Age   | Max Age | Min Age |
| ------------- | ------- | ------- |
| 21.14         | 58      | 9       |


From calculating summary statistics and reading the data set, I noticed issues such as the use of other and prefer not to say in the gender variable which can provide inaccuracies during analysis since some players did not explicitly list their gender. Additionally, the length of hashedEmail is difficult for the use of identification. Furthermore, from just looking at the data set, I am unable to determine the accuracy of classification regarding experience level and whether their method was effective. Also, the expression of time played in hours is not as accurate as it could be by using minutes instead.

In [10]:
# Read the sessions data set.
sessions <- read_csv("data/sessions.csv")
sessions

Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


hashedEmail                                                     
1    bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
2    36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
3    f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
4    bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
5    36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
6    bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
7    fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
8    ad6390295640af1ed0e45ffc58a53b2d9074b0eea694b16210addd44d7c81f83
9    96e190b0bf3923cd8d349eee467c09d1130af143335779251492eb4c2c058a5f
10   36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
11   fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
12   f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
13   f2826fb8dbce4d450348f99cb27ade184b713998d9679780442efaaf218038f2
14   b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
15   24d4892c2c3ce11d3e54c3bf31ee218901cac7a2f564911d67424c8fdf0874fe
16   fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
17   88247d9a46fc214a12485dcbcbb03a8ddebfe8c1ec5fe209cd660147625c8e62
18   f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d
19   36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
20   fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
21   bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
22   36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
23   fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
24   ad6390295640af1ed0e45ffc58a53b2d9074b0eea694b16210addd44d7c81f83
25   ad6390295640af1ed0e45ffc58a53b2d9074b0eea694b16210addd44d7c81f83
26   fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
27   36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
28   bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
29   b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
30   fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
⋮    ⋮                                                               
1506 ad6390295640af1ed0e45ffc58a53b2d9074b0eea694b16210addd44d7c81f83
1507 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
1508 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
1509 b3510c708bd50bf9f75e6e02bb6fe14edb705e0ea671ee769a45c613750e8b58
1510 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
1511 e74c60a92c0100e7240be56d66969db85856152b048c639331a3595f901461ae
1512 b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
1513 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
1514 e74c60a92c0100e7240be56d66969db85856152b048c639331a3595f901461ae
1515 fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
1516 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
1517 a1e0438d568b9c05a93cadaf99969e4d44794d19dea31edd723cbc916f878c33
1518 fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
1519 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
1520 55d24216db39c27e1f17cc43d3127cbf8ed76ada6d098202b53ded319855e2c1
1521 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
1522 b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
1523 b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
1524 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
1525 fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
1526 18936844e06b6c7871dce06384e2d142dd86756941641ef39cf40a9967ea14e3
1527 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
1528 18936844e06b6c7871dce06384e2d142dd86756941641ef39cf40a9967ea14e3
1529 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
1530 b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
1531 36d9cbb4c6

In [11]:
count(sessions)

n   
1 1535

In [12]:
# Provide summary statistics for the sessions data set by finding the number of session occurrences by email.
sessions_email <- sessions |>
    group_by(hashedEmail) |>
    count()
sessions_email

sessions_email_max <- sessions_email |>
    ungroup() |>
    summarize(n = max(n))
sessions_email_max

hashedEmail                                                      n  
1   0088b5e134c3f0498a18c7ea6b8d77b4b0ff1636fc93355ccc95b45423367832   2
2   060aca80f8cfbf1c91553a72f4d5ec8034764b05ab59fe8e1cf0eee9a7b67967   1
3   0ce7bfa910d47fc91f21a7b3acd8f33bde6db57912ce0290fa0437ce0b97f387   1
4   0d4d71be33e2bc7266ee4983002bd930f69d304288a8663529c875f40f1750f3  13
5   0d70dd9cac34d646c810b1846fe6a85b9e288a76f5dcab9c1ff1a0e7ca200b3a   2
6   11006065e9412650e99eea4a4aaaf0399bc338006f85e80cc82d18b49f0e2aa4   1
7   119f01b9877fc5ea0073d05602a353b91c4b48e4cf02f42bb8d661b46a34b760   1
8   18936844e06b6c7871dce06384e2d142dd86756941641ef39cf40a9967ea14e3  41
9   1a2b92f18f36b0b59b41d648d10a9b8b20a2adff550ddbcb8cec2f47d4d881d0   1
10  1d2371d8a35c8831034b25bda8764539ab7db0f63938696917c447128a2540dd   1
11  200db0417ee70863c7e2a0f214c737476fb86de5869d0b907a077d726357e687   1
12  20ae3f8b5323674fe6e537bc91928841cfc2a00c1b36522fd25370bb6fa89d36   2
13  22a78bfd44e46931261d06864b92cf9a8fd77fb02bec66cb71646dd9da222a0d   2
14  23fe711e0e3b77f1da7aa221ab1192afe21648d47d2b4fa7a5a659ff443a0eb5   1
15  24d4892c2c3ce11d3e54c3bf31ee218901cac7a2f564911d67424c8fdf0874fe  79
16  25879aecc205544bc6505f9faf768356e0a3b7126057308ae3235f8a514e9a62   1
17  2917e2dacafed92d017b9f356ce90a442bea6e4ad59dbbb484a920398acb8974   1
18  2cfed571797b66cc810c32562fc5b0f70b5bec0f5250799245527135bbf9662e   1
19  336adf9b9070abb44eaec22a48b04d8cda6cf0c98a46eef37779a1dcb6f5b4bb   3
20  36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686 130
21  3caa832978e0596779f4ee7c686c4592fb6de893925025d25bf26be9a8cd03f5   1
22  42382bb2af563e6979c5b84d288cbb49bb2ef90b1bcf196c2902da4ce20eb7f3   1
23  42eafe96ed5c1684e3b5cc614d1b01a117173d3ec6898abcf1177d9822bd39f7  11
24  4b01bce3f141289709e8278b02ba5d2aaa7105d7ccb9c7deb37670a80e332774   1
25  4bed6f49b672974f706aa7f020b4f29020326bf594520593557cd381a51559b8   1
26  4bfad3613c71ace05644bf210195d9fb0d3d9513753ad2f8a16d1a80699bb906   1
27  55d24216db39c27e1f17cc43d3127cbf8ed76ada6d098202b53ded319855e2c1   4
28  577aa5f15468252b1c6f32dcd515012923476292e30f95e6b4128c770e993f92   1
29  5a340c0e3d1aa3e579efc625bd3e5bca7fc25f7115b68e9411f457464bfd3df4   2
30  5afb9b4ba7f0bb42d4e4d72ee21be8a51131a7677f0519b4606fdbf770eec750   1
⋮   ⋮                                                                ⋮  
96  d48302dee689942b60f7e9756eba99d0be030df0ab6f553904a9dbbc44d0b5d5   1
97  d51a8c57269fe347b4b7760ec29c420832cb36a11c47569a9200872831910c05   1
98  d572f391d452b76ea2d7e5e53a3d38bfd7499c7399db299bd4fedb06a46ad5bb   1
99  d9473710057f7d42f36570f0be83817a4eea614029ff90cf50d8889cdd729d11   1
100 dbb20960cd4db4900dde7832e349dae46f7688583c8138b34785d9c7e509b5b9   1
101 dc73467f73263dd4a07838330dd1cc115aa3f8b03538911ee6aac048405939c4   1
102 e21a324ccf5c873bafe82e47d5137b36aa312ee4803eeb5cbc6cf945f9bb3f38   1
103 e3175c379bbb83c8786d2dd719cf844575aca24d2c81d643dc7f12f6227becf7   1
104 e3e1bef1ce259c995332635c0ec630a575713451cbe742d7ca93e4c03d0826df  12
105 e44041459da2102dc20147ed6f0db4753547be66fc4ddea037bafca64004de2d   1
106 e47ae138603b9a7d6e9b97554bac2daf0477b30e0a4f29ed9d3deacc3c7ec155   1
107 e74c60a92c0100e7240be56d66969db85856152b048c639331a3595f901461ae  47
108 ed3e6d043dee74fad932be44d0600f4502055198c511927b0c906f50b57d9582   1
109 f19e136ddde68f365afc860c725ccff54307dedd13968e896a9f890c40aea436   6
110 f1b432523542f90c61176a555ccb2144468d76c91a32d74082ab8c101f9d25b6   1
111 f2826fb8dbce4d450348f99cb27ade184b713998d9679780442efaaf218038f2   7
112 f3c813577c458ba0dfef80996f8f32c93b6e8af1fa939732842f2312358a88e9   3
113 f459ce1862ec9a43b6c714810390dfd632ed104eb11378b3a2a134b1b107bc75   2
114 f4f29f2465743ead8c3ce02fc84542afe33ae3c95ef56a42b8f7785550fe9e4a   2
115 f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d  27
116 f707d141be44b2bf963eb171dbd14599059b4dda09aa002ba95ead4fa67273a7   1
117 f7875ae87a61632030d5c4029ee8cf081be7047b2b4a9c04ab5a706f59786197   1
118 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42

n  
1 310

In [19]:
# Provide summmary statistics by determining if any sessions started or ended at the same time and the number that did.
sessions_start <- sessions |>
    group_by(start_time) |>
    count()
sessions_start

sessions_start_max <- sessions_start |>
    ungroup() |>
    summarize(n = max(n))
sessions_start_max

sessions_end <- sessions |>
    group_by(end_time) |>
    count()
sessions_end

sessions_end_max <- sessions_end |>
    ungroup() |>
    summarize(n = max(n))
sessions_end_max

start_time       n
1    01/05/2024 00:58 1
2    01/05/2024 02:28 1
3    01/05/2024 02:41 1
4    01/05/2024 03:29 1
5    01/05/2024 04:54 1
6    01/05/2024 05:46 1
7    01/05/2024 18:16 1
8    01/05/2024 20:34 1
9    01/05/2024 22:42 1
10   01/05/2024 23:46 1
11   01/06/2024 00:27 1
12   01/06/2024 02:22 1
13   01/06/2024 02:29 1
14   01/06/2024 05:42 1
15   01/06/2024 05:45 1
16   01/06/2024 06:22 1
17   01/06/2024 06:35 1
18   01/06/2024 14:44 1
19   01/06/2024 17:01 1
20   01/06/2024 17:29 1
21   01/06/2024 17:32 1
22   01/06/2024 18:11 1
23   01/06/2024 18:27 1
24   01/06/2024 18:28 1
25   01/06/2024 20:01 1
26   01/06/2024 22:19 1
27   01/06/2024 23:51 1
28   01/07/2024 00:10 1
29   01/07/2024 00:55 1
30   01/07/2024 01:43 1
⋮    ⋮                ⋮
1475 30/08/2024 03:40 1
1476 30/08/2024 04:05 1
1477 30/08/2024 05:04 1
1478 30/08/2024 05:08 1
1479 30/08/2024 15:25 1
1480 30/08/2024 15:48 1
1481 30/08/2024 16:18 1
1482 30/08/2024 17:49 1
1483 30/08/2024 21:36 1
1484 30/08/2024 23:47 1
1485 31/05/2024 00:36 1
1486 31/05/2024 02:29 1
1487 31/05/2024 02:41 1
1488 31/05/2024 04:44 1
1489 31/05/2024 04:56 1
1490 31/05/2024 21:28 1
1491 31/07/2024 02:27 1
1492 31/07/2024 02:58 1
1493 31/07/2024 03:23 1
1494 31/07/2024 03:24 1
1495 31/07/2024 03:35 1
1496 31/07/2024 04:01 1
1497 31/07/2024 04:05 1
1498 31/07/2024 07:20 1
1499 31/08/2024 02:10 1
1500 31/08/2024 02:29 1
1501 31/08/2024 03:14 1
1502 31/08/2024 16:07 1
1503 31/08/2024 22:44 1
1504 31/08/2024 23:35 1

n
1 3

end_time         n
1    01/05/2024 01:24 1
2    01/05/2024 02:41 1
3    01/05/2024 03:24 1
4    01/05/2024 03:50 1
5    01/05/2024 04:38 1
6    01/05/2024 06:17 1
7    01/05/2024 06:48 1
8    01/05/2024 18:21 1
9    01/05/2024 20:54 1
10   01/05/2024 22:54 1
11   01/06/2024 01:46 1
12   01/06/2024 03:37 1
13   01/06/2024 04:00 1
14   01/06/2024 06:31 1
15   01/06/2024 06:43 1
16   01/06/2024 06:57 1
17   01/06/2024 08:34 1
18   01/06/2024 16:45 1
19   01/06/2024 17:43 1
20   01/06/2024 18:12 1
21   01/06/2024 18:28 1
22   01/06/2024 18:36 1
23   01/06/2024 18:37 1
24   01/06/2024 18:46 1
25   01/06/2024 21:32 1
26   01/06/2024 22:30 1
27   01/06/2024 23:55 1
28   01/07/2024 01:03 1
29   01/07/2024 01:46 1
30   01/07/2024 03:33 1
⋮    ⋮                ⋮
1461 30/08/2024 04:58 1
1462 30/08/2024 06:09 1
1463 30/08/2024 06:10 1
1464 30/08/2024 15:55 1
1465 30/08/2024 16:38 1
1466 30/08/2024 18:04 1
1467 30/08/2024 18:48 1
1468 31/05/2024 01:40 1
1469 31/05/2024 03:39 1
1470 31/05/2024 04:05 1
1471 31/05/2024 04:49 1
1472 31/05/2024 06:30 1
1473 31/05/2024 23:04 1
1474 31/07/2024 00:37 1
1475 31/07/2024 03:18 1
1476 31/07/2024 03:21 1
1477 31/07/2024 04:17 1
1478 31/07/2024 04:35 1
1479 31/07/2024 05:21 1
1480 31/07/2024 05:23 1
1481 31/07/2024 05:29 1
1482 31/07/2024 07:26 1
1483 31/08/2024 01:13 1
1484 31/08/2024 01:14 1
1485 31/08/2024 03:12 1
1486 31/08/2024 03:59 1
1487 31/08/2024 06:14 1
1488 31/08/2024 16:14 1
1489 31/08/2024 23:20 1
1490 NA               2

n
1 3

In [21]:
# Calculate the average start and end time for sessions using the original times.
avg_original_start <- sessions |>
    summarize(original_start_time = mean(original_start_time))
avg_original_start

avg_original_end <- sessions |>
    summarize(original_end_time = mean(original_end_time, na.rm = TRUE))
avg_original_end

original_start_time
1 1.719201e+12

original_end_time
1 1.719196e+12

### Sessions Data Set
The sessions data set includes a list of individual play sessions by each player with data on the session. In the data set, there are 5 variables, as both character and double data types, and 1535 observations. 
- hashedEmail(chr): an encrypted version of an email address used to protect the user's personal information, used in coordination with the players data set
- start_time(chr): the time the Minecraft session began in day, month, year, time format
- end_time(chr): the time the Minecraft session ended in day, month, year, time format
- original_start_time(dbl): the start time of the session as a single number
- original_end_time(dbl): the end time of the session as a single number

**Hashed Email**

By using group_by and count, I am able to determine the number of sessions each player participated in. Out of the 196 total players, 125 are present in the sessions data set, where the most sessions played by a single player(identified by their hashedEmail) is 310 sessions. 

**Session Start and End Time**

From calculating summary statistics on the start and end time of sessions, I found that there are only 1504 observations for start time and 1490 for end time. This indicates that several sessions started or ended at the same time, the max being 3, or in some situations the time was listed as NA.

**Original Start and End Time**

The average original start and end times are shown in the table below.
| Average Original Start Time | Average Original End Time |
| --------------------------- | ------------------------- |
| 1.72e+12                    | 1.72e+12                  |

After performing summary statistics on the sessions data set, I noticed that the number of hashedEmails identified is less than the total players. With this, I assume that 71 players did not participate in a session, however, it is unclear the exact reason given only the data. Additionally, I am unsure as to how the original start and end times were converted and whether they are accurate.